types/trouter/index.d.ts

```diff
@@ -0,0 +1,47 @@
+// Type definitions for trouter 2.0
+// Project: https://github.com/lukeed/trouter
+// Definitions by: Markus Lanz <https://github.com/stahlstift>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.3
+
+declare namespace Trouter {
+    interface FindResult<T> {
+        params: { [k: string]: string; };
+        handlers: T[];
+    }
+
+    type ValidMethods = 'get' | 'head' | 'patch' | 'options' |
+        'connect' | 'delete' | 'trace' | 'post' | 'put';
+
+    type FindReturn<T> = FindResult<T> | false;
+}
+
+declare class Trouter<T = any> {
+    constructor();
+
+    find(method: Trouter.ValidMethods, url: string): Trouter.FindReturn<T>;
+
+    add(method: Trouter.ValidMethods, pattern: string, ...handlers: T[]): this;
+
+    all(pattern: string, ...handlers: T[]): this;
```

Comment:
Not important, but I feel like there ought to be a way to move all of these into a shared definition... possibly extending `add`? 

Something like [this](https://github.com/lukeed/httpie/blob/master/httpie.d.ts#L19-L27) but for classes, of course.

Reply:
@SimonSchick 
I don't know if that is possible. 
Do you know?